File: src/main/java/nars/main_nogui/CommandLineParameters.java
package nars.main_nogui;

import kotlin.jvm.JvmStatic;

/**
 * The parameters used when the system is invoked from command line
 */
public class CommandLineParameters {

    /**
     * Decode the silence level
     *
     * @param args Given arguments
     * @param r    The corresponding reasoner
     */@JvmStatic
    public static void decode(String[] args, ReasonerBatch r) {
        for (var i = 0; i < args.length; i++) {
            var arg = args[i];
            if ("--silence".equals(arg)) {
                ++i;
                arg = args[i];
                r.getSilenceValue().set(Integer.parseInt(arg));
            }
        }
    }

}

File: src/main/java/nars/main_nogui/CommandLineParameters.kt
package nars.main_nogui

/**
 * The parameters used when the system is invoked from command line
 */
object CommandLineParameters {
    /**
     * Decode the silence level
     *
     * @param args Given arguments
     * @param r    The corresponding reasoner
     */
    @JvmStatic
    fun decode(args: Array<String>, r: ReasonerBatch) {
        var i = 0
        while (i < args.size) {
            var arg = args[i]
            if ("--silence" == arg) {
                ++i
                arg = args[i]
                r.silenceValue.set(Integer.parseInt(arg))
            }
            i++
        }
    }
}